SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorLayoutDetails.cpp

```diff
@@ -83,3 +113,128 @@ FReply FSpatialGDKEditorLayoutDetails::ClickedOnButton()
+	const FString IOSProjectPath = FString::Printf(TEXT("../../../%s/%s.uproject"), *ProjectName, *ProjectName);
+	FString TravelUrl;
+	FString SpatialOSCommandLineArgs = FString::Printf(TEXT("-workerType %s"), *(SpatialGDKSettings->MobileWorkerType));
+	if (SpatialGDKSettings->bMobileConnectToLocalDeployment)
+	{
+		TravelUrl = SpatialGDKSettings->MobileRuntimeIP;
+	}
+	else
+	{
+		TravelUrl = TEXT("connect.to.spatialos");
+
+		if (SpatialGDKSettings->DevelopmentAuthenticationToken.IsEmpty())
+		{
+			FReply GeneratedToken = GenerateDevAuthToken();
+			if (!GeneratedToken.IsEventHandled())
+			{
+				return false;
+			}
+		}
+
+		SpatialOSCommandLineArgs += FString::Printf(TEXT(" +devauthToken %s"), *(SpatialGDKSettings->DevelopmentAuthenticationToken));
+		if (!SpatialGDKSettings->DevelopmentDeploymentToConnect.IsEmpty())
+		{
+			SpatialOSCommandLineArgs += FString::Printf(TEXT(" +deployment %s"), *(SpatialGDKSettings->DevelopmentDeploymentToConnect));
+		}
+	}
+
+	const FString FinalCommandLineArgs = FString::Printf(TEXT("%s %s %s %s"), *IOSProjectPath, *TravelUrl, *SpatialOSCommandLineArgs, *(SpatialGDKSettings->MobileExtraCommandLineArgs));
+	CommandLineArgsFile = FPaths::ConvertRelativePathToFull(FPaths::Combine(*FPaths::ProjectLogDir(), TEXT("ue4commandline.txt")));
+
+	if (!FFileHelper::SaveStringToFile(FinalCommandLineArgs, *CommandLineArgsFile, FFileHelper::EEncodingOptions::ForceUTF8WithoutBOM))
+	{
+		UE_LOG(LogSpatialGDKEditorLayoutDetails, Error, TEXT("Failed to write command line args to file: %s"), *CommandLineArgsFile);
+		FMessageDialog::Open(EAppMsgType::Ok, FText::FromString(FString::Printf(TEXT("Failed to write command line args to file: %s"), *CommandLineArgsFile)));
+		return false;
+	}
+
+	return true;
+}
+
+bool FSpatialGDKEditorLayoutDetails::TryPushCommandLineArgs(const FString Executable, const FString ExeArguments, const FString CommandLineArgsFile)
+{
+	FString OutCommandLineArgsFile;
+	FString ExeOutput;
+	FString StdErr;
+	int32 ExitCode;
+
+	FPlatformProcess::ExecProcess(*Executable, *ExeArguments, &ExitCode, &ExeOutput, &StdErr);
+	if (ExitCode != 0)
+	{
+		UE_LOG(LogSpatialGDKEditorLayoutDetails, Error, TEXT("Failed to update the mobile client. %s %s"), *ExeOutput, *StdErr);
+		FMessageDialog::Open(EAppMsgType::Ok, FText::FromString(TEXT("Failed to update the mobile client. See the Output log for more information.")));
+		return false;
+	}
+
+	UE_LOG(LogSpatialGDKEditorLayoutDetails, Log, TEXT("Successfully stored command line args on device: %s"), *ExeOutput);
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	if (!PlatformFile.DeleteFile(*CommandLineArgsFile))
+	{
+		UE_LOG(LogSpatialGDKEditorLayoutDetails, Error, TEXT("Failed to delete file %s"), *CommandLineArgsFile);
+		FMessageDialog::Open(EAppMsgType::Ok, FText::FromString(FString::Printf(TEXT("Failed to delete file %s"), *CommandLineArgsFile)));
+	}
```

Comment:
Should return false here.

Reply:
What's the behavior of DeleteFile if there is no file? Does it fail?

Reply:
DeleteFile will return false in that case: https://docs.unrealengine.com/en-US/API/Runtime/Core/GenericPlatform/IPlatformFile/DeleteFile/index.html

and yes, will add the `return false` :)